radixdlt-engine/src/main/java/com/radixdlt/atom/TxLowLevelBuilder.java

```diff
@@ -159,37 +145,34 @@ public void instruction(REInstruction.REMicroOp op, byte[] data) {
     }
   }
 
-  public TxLowLevelBuilder message(byte[] bytes) throws MessageTooLongException {
-    if (bytes.length > 255) {
-      throw new MessageTooLongException(bytes.length);
+  public TxLowLevelBuilder message(byte[] bytes, int limit) throws MessageTooLongException {
+    if (bytes.length > limit) {
+      throw new MessageTooLongException(limit, bytes.length);
     }
 
-    instruction(REInstruction.REMicroOp.MSG, varLengthData(bytes));
-    return this;
-  }
+    var buf = ByteBuffer.allocate(Short.BYTES + bytes.length);
+    buf.putShort((short) bytes.length);
```

Comment:
Do we need to protect against the possibility someone ups the limit above 32,767 (max short)?

Perhaps we artificially cap the limit to max short at the top of this method?

Reply:
Do we ever ensure the limit is <= 32,767?

Reply:
No, there are no checks for that. We don't have it because it should be part of the validation of `RERulesConfig`, which is missing. Given that `RERulesConfig` instances are always hardcoded, I'm uncertain if we should have such a validation in the first place.

Reply:
I'm comfortable with this for now, assuming this would throw at the cast?

Reply:
No, it wouldn't. Added dedicated check to ensure length never exceed "hardware" limit.